test/unit/client.js

```diff
@@ -3,6 +3,35 @@ import { StreamChat } from '../../src/client';
 
 const expect = chai.expect;
 
+describe('StreamChat getInstance', () => {
+	beforeEach(() => {
+		delete StreamChat._instance;
+	});
+
+	it('instance is stored as static property', function () {
+		expect(StreamChat._instance).to.be.undefined;
+
+		const client = StreamChat.getInstance('key');
+		expect(client).to.equal(StreamChat._instance);
+	});
+
+	it('always return the same instance', function () {
+		const client1 = StreamChat.getInstance('key1');
+		const client2 = StreamChat.getInstance('key1');
+		const client3 = StreamChat.getInstance('key1');
+		expect(client1).to.equal(client2);
+		expect(client2).to.equal(client3);
+	});
+
+	it('changin params has no effect', function () {
+		const client1 = StreamChat.getInstance('key2');
+		const client2 = StreamChat.getInstance('key3');
+
+		expect(client1).to.equal(client2);
+		expect(client2.key).to.eql('key2');
+	});
```

Comment:
Can we also add following scenario to unit-test?

```js
const client1 = StreamChat.getInstance('key2');
await client1.setUser({ id: 'vishal });
const client2 = StreamChat.getInstance('key3'); // or key3
await client2.setUser({ id: 'vishal });

```

Reply:
Currently if you call setUser twice, it throws following error:

```
'Use client.disconnect() before trying to connect as a different user. connectUser was called twice.'
```

But I think we should allow it if current user is same as id provided to `setUser`

Reply:
pushed a test https://github.com/GetStream/stream-chat-js/pull/599/commits/d2fcfc3394cdc0501db4dc7e3aa5f7e4363324e9

Reply:
@vishalnarkhede yea I actually put a comment in the PR description about this which is out of the scope of this PR. This is still an integration mistake but we can make it easier for users by not throwing error

Reply:
Ahaa ... sorry missed it!!

Can we add it though ... seems like a simple check

Reply:
yea totally will open another PR on top of this one